atlasdb-tests-shared/src/test/java/com/palantir/timelock/paxos/InMemoryTimelockServicesTest.java

```diff
@@ -59,7 +62,31 @@ public void tearDown() {
     public void canGetTimestamp() {
         long ts1 = timestampService.getFreshTimestamp();
         long ts2 = timelockService.getFreshTimestamp();
-        assertThat(ts1).isLessThan(ts2);
+        testTs(ts1, ts2);
+    }
+
+    @Test
+    public void timestampsAreConsistent() {
+        long ts1 = delegatingTimelockService.getFreshTimestamps(1).getLowerBound();
+        long ts2 = timestampService.getFreshTimestamp();
+        testTs(ts1, ts2);
+
+        long ts3 = delegatingTimelockService.getFreshTimestamps(1).getLowerBound();
+        testTs(ts2, ts3);
+
+        long ts4 = delegatingTimelockService
+                .startIdentifiedAtlasDbTransactionBatch(1)
+                .get(0)
+                .startTimestampAndPartition()
+                .timestamp();
+        testTs(ts3, ts4);
+
+        long ts5 = delegatingTimelockService.getFreshTimestamp();
+        testTs(ts4, ts5);
+    }
+
+    private void testTs(long ts2, long ts3) {
+        assertThat(ts2).isLessThan(ts3);
```

Comment:
Also for naming: this should probably be: `assertTimestampsAreOrdered` or something, with parameters and variables throughout `timestamp1`, `timestamp2` instead of `tsX`.

Reply:
Also, is there any reason why we can't make this stronger - fast forward the timestamp, and assert that they're the same?

Reply:
Maybe we don't want to commit to that (and only want to commit to the fact that they are monotonically increasing).

Reply:
yeah, there's a separate fast forward test; for this one I just wanted to make sure they're monotonically increasing if we shuffle which methods to use.